plugins/thunderbolt/fu-thunderbolt-device.c

```diff
@@ -320,6 +320,75 @@ fu_thunderbolt_device_get_attr_uint16(FuThunderboltDevice *self, const gchar *na
 	return (guint16)val;
 }
 
+static gboolean
+fu_thunderbolt_device_set_port_offline(FuUdevDevice *device)
+{
+	g_autoptr(GError) error_local_offline = NULL;
+	g_autoptr(GError) error_local_rescan = NULL;
+
+	if (!fu_udev_device_write_sysfs(device, "usb4_port1/offline", "1", &error_local_offline)) {
+		g_warning("Setting port offline failed: %s", error_local_offline->message);
+		return FALSE;
+	}
+	if (!fu_udev_device_write_sysfs(device, "usb4_port1/rescan", "1", &error_local_rescan)) {
+		g_warning("Rescan on port failed: %s", error_local_rescan->message);
+		return FALSE;
+	}
```

Comment:
these should be critical failures or not?  It seems to me they should be, you should have a parameter of error to the function and be setting it when you return false instead of leaving a warning.

Reply:
fixed in https://github.com/fwupd/fwupd/pull/4046/commits/02304c8e6ecd9c8484696bb959507e81b706efb8